feat: Add Issue section to PR template (#4259)
Codecademy/docs: .github/PULL_REQUEST_TEMPLATE.md

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

> # Description

Author: ```suggestion
### Description
```

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

> ## Issue Solved

Author: ```suggestion
### Issue Solved
```

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

### Issue Solved

<!--

> Please use the following format(s) to link the issue numbers that your contribution addresses: Fixes #issue number(e.g., #123)
> Relates to #issue number(e.g., #123)

Reviewer: ```suggestion
Please use the following format(s) to link the issue numbers that your contribution addresses: 
Closes #issue-number (where `issue-number` corresponds to the PRs' respective issue)
```
